hazelcast/src/main/java/com/hazelcast/internal/diagnostics/DiagnosticsLogFile.java

```diff
@@ -107,36 +104,22 @@ private void renderStaticPlugins() throws IOException {
     }
 
     private void renderPlugin(DiagnosticsPlugin plugin) throws IOException {
-        logWriter.clean();
-        plugin.run(logWriter);
-        //bufferedWriter.append()
-        int desiredLength = charBuff.length;
-        int actualSize = logWriter.length();
-        while (desiredLength < actualSize) {
-            desiredLength *= 2;
-        }
+        logWriter.init(printWriter);
 
-        if (desiredLength != charBuff.length) {
-            charBuff = new char[desiredLength];
-        }
-
-        logWriter.copyInto(charBuff);
-        bufferedWriter.write(charBuff, 0, actualSize);
-        fileLength += actualSize;
+        plugin.run(logWriter);
     }
 
-    private BufferedWriter newWriter() throws FileNotFoundException {
+    private PrintWriter newWriter() throws FileNotFoundException {
         FileOutputStream fos = new FileOutputStream(file, true);
         CharsetEncoder encoder = Charset.forName("UTF-8").newEncoder();
-        return new BufferedWriter(new OutputStreamWriter(fos, encoder));
+        return new PrintWriter(new BufferedWriter(new OutputStreamWriter(fos, encoder), Short.MAX_VALUE));
```

Comment:
Okay, that's fine.

For the `close()` propagation you are right for the `PrintWriter` and the `OutputStreamWriter` (which closes its encoder which closes the `FileOutputStream`). But as I already said, the `BufferedWriter` doesn't seem to propagate the `close()` call, it just flushes and sets the reference to `null`. Do we need to do a manual close to release the `OutputStreamWriter` correctly?